contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,124 @@
+    uint96 id; // grant ID, as uint96 to pack into same slot as owner (this implies a max of 2^96-1 = 7.9e28 grants)
+    address owner; // grant owner (has permissions to modify grant information)
+    address payee; // address that receives funds donated to this grant
+    string metaPtr; // URL pointing to grant metadata (for off-chain use)
   }
 
-  function activate(bytes32 _id) external onlyOwner {
-    grants[_id].state = State.ACTIVE;
-  }
+  /// @notice Mapping from Grant ID to grant data
+  mapping(uint96 => Grant) public grants; // TODO use an array instead with ID to index it? Which is better? Will array copy full array to memory in `getAllGrants`?
 
-  function closed(bytes32 _id) public onlyOwner {
-    grants[_id].state = State.CLOSED;
-  }
+  // --- Events ---
+  /// @notice Emitted when a new grant is created
+  event GrantCreated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
+
+  /// @notice Emitted when a grant's owner is changed
+  event GrantUpdated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
 
-  function isConfig(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CONFIG;
+  // --- Core methods ---
+  /**
+   * @notice Create a new grant in the registry
+   * @param _owner Grant owner (has permissions to modify grant information)
+   * @param _payee Address that receives funds donated to this grant
+   * @param _metaPtr URL pointing to grant metadata (for off-chain use)
+   */
+  function createGrant(
+    address _owner,
+    address _payee,
+    string memory _metaPtr
+  ) external {
+    uint96 _id = grantCount;
+    grants[_id] = Grant(_id, _owner, _payee, _metaPtr);
+    emit GrantCreated(_id, _owner, _payee, _metaPtr);
+    grantCount += 1;
   }
 
-  function isActive(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.ACTIVE;
+  /**
+   * @notice Update the owner of a grant
+   * @param _id ID of grant to update
+   * @param _owner New owner address
+   */
+  function updateGrantOwner(uint96 _id, address _owner) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.owner = _owner;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
-  function isClosed(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CLOSED;
+  /**
+   * @notice Update the payee of a grant
+   * @param _id ID of grant to update
+   * @param _payee New payee address
+   */
+  function updateGrantPayee(uint96 _id, address _payee) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.payee = _payee;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
-  function getGrantCount() public view returns (uint256) {
-    return grantCount;
+  /**
+   * @notice Update the payee of a grant
```

Comment:
Notice is incorrect due to copy/paste

Reply:
Fixed in 756f836a84c1f1b47bf58f12734df5e871fd6b50

Reply:
Fixed in 756f836a84c1f1b47bf58f12734df5e871fd6b50